sklearn/gaussian_process/gpc.py

```diff
@@ -74,7 +73,7 @@ def optimizer(obj_func, initial_theta, bounds):
                 # the corresponding value of the target function.
                 return theta_opt, func_min
 
-        Per default, the 'fmin_l_bfgs_b' algorithm from scipy.optimize
+        Per default, the 'L-BFGS-B' algorithm from scipy.optimize.maximize
         is used. If None is passed, the kernel's parameters are kept fixed.
         Available internal optimizers are::
 
```

Comment:
Should we then deprecate `fmin_l_bfgs_b` as the input value in favor of `L-BFGS-B`, or `lbfgs`?

Reply:
Yes, I think that would be in order, and similarly I wanted to see if allowing other scipy optimizers in Gaussian processes would be interesting.  Though I would rather do that in a follow-up PR, and keep this as a minimal refactoring not affecting backward compatibility.

Reply:
I agree allowing other optimizers would be a different PR, but since you're touching the docstring here, it makes sense for the accepted value to be the same or similar to what you mention in the docstring, I think.

But if you wanna do the deprecation in a different PR, I'm happy with that as well, and then this LGTM.

Reply:
I haven't changed the docstring value [here](https://github.com/scikit-learn/scikit-learn/pull/14250/files#diff-e5adf824b9db41efb7aed6b3c4a69820R464) only changed that the algorithm is called `L-BFGS-B`, not `fmin_l_bfgs_b`. That fix would apply even before this PR as `fmin_l_bfgs_b` is not an algorithm name it's the scipy function name for that optimizer. 

Will do the deprecated in a follow up PR :) Thanks for the review!